### ISSUE TITLE
no-trailing-whitespace ignores trailing whitespace in comments

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#2049]
### Bug Report

- __TSLint version__: current master
- __TypeScript version__: 2.1.4
- __Running TSLint via__: cli

#### TypeScript code being linted

```ts
// comment with trailing spaces    
/* multiline comment 
    with trailing whitespace in first line */
```

with `tslint.json` configuration:

```json
"no-trailing-whitespace": true
```

#### Actual behavior

no error

#### Expected behavior

```ts
// comment with trailing spaces    
                               ~~~~ [trailing whitespace]
/* multiline comment 
                    ~ [trailing whitespace]
   with trailing whitespace in first line */
```

Is this intended or can this be fixed? And should this be configurable with a config option?

The fix would not be difficult. I'd like to fix this if you want it

**Comment by nchen63** [comment]
Is there ever a valid reason for that? I'd prefer not to add an option if not

**Comment by Martinspire** [comment]
Why would you want to notify trailing whitespace in comments? Isn't the whole point of comments to not run any validations on it other than it being opened/closed properly?

**Comment by adidahiya** [comment]
> Isn't the whole point of comments to not run any validations on it other than it being opened/closed properly?

I guess this is open to interpretation, but no, I don't think that's the _whole point_ of comments. I don't see why you would want trailing whitespace in some part of your code and not others. If you and others feel strongly about it, you could send a PR to allow the rule to ignore comments.

**Comment by JoshuaKGoldberg** [comment]
^ this. I like having trailing whitespace in my TSDoc comments, but otherwise it's a waste.

```typescript
/**
 *.{description}
 *.
 *.{params,etc}
 */
```

Config for the rule might want to include whether all comments are ignored or just TSDoc comments.
